// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_outer_classname = "StatisticsProto";
option objc_class_prefix = "STS";

package statistics;

// The greeting service definition.
service Greeter {
  rpc GetStatistics (StatisticRequest) returns (StatisticResponse) {};

  rpc TopPosts (TopPostsRequest) returns(TopPostsResponse) {};

  rpc TopUsers (TopUsersRequest) returns(TopUsersResponse) {};
}

message Post {
  int64 post_id = 1;
  string author = 2;
  string description = 3;
  bytes image = 4;
  int64 likes = 5;
  int64 views = 6;
}

message User {
  int64 likes = 1;
  string author = 2;
}

message StatisticRequest {
  int64 post_id = 1;
}

message StatisticResponse {
  int64 likes = 1;
  int64 views = 2;
}

message TopPostsRequest {
  string filter = 1;
}

message TopPostsResponse {
  repeated Post posts = 1;
}

message TopUsersRequest {}

message TopUsersResponse {
  repeated User users = 1;
}